mountpoint-s3/src/fs.rs

```diff
@@ -764,19 +769,77 @@ where
         };
 
         if offset != dir_handle.offset() {
+            // POSIX allows seeking an open directory. That's a pain for us since we are streaming
+            // the directory entries and don't want to keep them all in memory. But one common case
+            // we've seen (https://github.com/awslabs/mountpoint-s3/issues/477) is applications that
+            // request offset 0 twice in a row. So we remember the last response and, if repeated,
+            // we return it again.
+            let last_response = dir_handle.last_response.lock().await;
+            if let Some((last_offset, entries)) = &*last_response {
```

Comment:
I had to do a bit of juggling in my head to figure out what `&*last_response` is doing: de-referencing the guard to get the value, then taking only the reference to that value.

Is it clearer to use Option's `as_ref()`? De-referencing of the guard behaves like normal (coerced), then we explicitly take a ref to the underlying values of the Option. https://doc.rust-lang.org/std/option/enum.Option.html#method.as_ref

```suggestion
            if let Some((last_offset, entries)) = last_response.as_ref() {
```

Reply:
Yeah, `&*` is just an old habit.